test/speech/index.js

```diff
@@ -0,0 +1,836 @@
+        assert.strictEqual(protoOpts.service, 'Speech');
+        assert.strictEqual(protoOpts.method, 'streamingRecognize');
+
+        return mockStream;
+      };
+
+      assert.strictEqual(speech.createRecognizeStream(request), mockStream);
+    });
+  });
+
+  describe('createDetectStream', function() {
+    var REQ = {};
+
+    it('should make the correct API request', function(done) {
+      var expected = 'how old is the Brooklyn Bridge';
+      var filePath = '../../system-test/data/bridge.raw';
+      var audioFile = fs.readFileSync(path.join(__dirname, filePath));
+      var request = REQ;
+      var upstreamWriteCount = 0;
+      var mockStream = {
+        write: function(data) {
+          upstreamWriteCount++;
+          assert(data);
+          if (upstreamWriteCount === 1) {
+            assert.deepEqual(data, {
+              streamingConfig: REQ
+            });
+            assert.strictEqual(data.streamingConfig, REQ);
+          } else if (upstreamWriteCount === 2) {
+            assert(data.audioContent);
+          }
+        },
+        on: function(event, handler) {
+          if (event === 'response') {
+            this.responseHandler = handler;
+          }
+        },
+        emit: function(event, data) {
+          if (event === 'data') {
+            this.pipeTo.write(data);
+          } else if (event === 'response') {
+            this.responseHandler(data);
+          }
+        },
+        once: function() {},
+        pipe: function(stream) {
+          this.pipeTo = stream;
+        }
+      };
+
+      speech.requestWritableStream = function(protoOpts) {
+        assert.strictEqual(protoOpts.service, 'Speech');
+        assert.strictEqual(protoOpts.method, 'streamingRecognize');
+
+        return mockStream;
+      };
+
+      var stream = speech.createDetectStream(request);
+      assert(stream !== mockStream);
+
+      stream.on('data', function(data) {
+        assert.deepEqual(data, [
+          {
+            transcript: expected
+          }
+        ]);
+        assert.equal(upstreamWriteCount, 2, 'two upstream writes');
+        done();
+      });
+
+      setTimeout(function() {
+        stream.write(audioFile);
+      }, 100);
+      setTimeout(function() {
+        mockStream.emit('response', 'response');
+        mockStream.emit('data', {});
+        mockStream.emit('data', {
+          results: [
+            {
+              alternatives: []
+            }
+          ]
+        });
+      }, 200);
+      setTimeout(function() {
+        mockStream.emit('data', {
+          results: [
+            {
+              alternatives: [
+                {
+                  transcript: expected
+                }
+              ]
+            }
+          ]
+        });
+      }, 300);
+    });
+  });
+
+  describe('operation', function() {
+    var REQ = {
+      name: '1234'
+    };
+
+    it('should make an operation instance', function() {
+      var operation = speech.operation('1234');
```

Comment:
I think for this we want to stub `Operation` and check that `operation` is an instance of it.